Adds custom styles for `<details>` and `<summary>` elements in Markdown content (#1735)
withastro/starlight: docs/src/content/docs/guides/authoring-content.md

  ```
  ````

## Details

Details (also known as “disclosures” or “accordion”) are useful to conceal information that is not immediately relevant to the user.

> The user can reveal further information with an interaction to toggle the visibility of the content.

Reviewer: This sentence was maybe saying the same thing twice a bit: “reveal further information” and “toggle the visibility of the content”.

I think we can maybe boil it down to this?

```suggestion
Users can click a short summary to expand and view the full content.
```

> Starlight does not provide a custom Markdown syntax for details, but you can use the standard HTML `<details>` and `<summary>` elements to create them while authoring your Markdown content.

Reviewer: I think we can describe the positive “what to do” here without mentioning what we don’t do. I do not _love_ “widget” as a word here, but it does appear to be what MDN uses, so maybe it’s OK? (And does work better than “element”, “component”, or “pattern”… any other ideas?)

```suggestion
Use the standard HTML [`<details>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details) and [`<summary>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/summary) elements in your Markdown content to create a disclosure widget.
```

> You can nest any other Markdown content types inside a `<details>` element.

Reviewer: Maybe “syntax” works better here?

```suggestion
You can nest any other Markdown syntax inside a `<details>` element.
```

> <details>
> <summary>Where and when the Andromeda constellation is best visible?</summary>
> 
> The [Andromeda constellation](<https://en.wikipedia.org/wiki/Andromeda_(constellation)>) is best visible in the night sky during the month of November at latitudes between `+90°` and `−40°`.
> 
> </details>
> 
> ```md
> <details>
> <summary>Where and when the Andromeda constellation is best visible?</summary>
> 
> The [Andromeda constellation](<https://en.wikipedia.org/wiki/Andromeda_(constellation)>) is best visible in the night sky during the month of November at latitudes between `+90°` and `−40°`.
> 
> </details>
> ```

Reviewer: Guess we need to make sure the demo content is also just right 😅 

Just a small English grammar detail here:

````suggestion
<details>
<summary>Where and when is the Andromeda constellation most visible?</summary>

The [Andromeda constellation](<https://en.wikipedia.org/wiki/Andromeda_(constellation)>) is most visible in the night sky during the month of November at latitudes between `+90°` and `−40°`.

</details>

```md
<details>
<summary>Where and when is the Andromeda constellation most visible?</summary>

The [Andromeda constellation](<https://en.wikipedia.org/wiki/Andromeda_(constellation)>) is most visible in the night sky during the month of November at latitudes between `+90°` and `−40°`.

</details>
```
````
